### PR TITLE
Chop computeReplicasForMetrics to smaller pieces

### PR DESCRIPTION
**Pull request by jbartosik**
Split computeReplicasForMetrics function into smaller pieces.

**Release note**:
```release-note
NONE
```

### CODE REVIEW

File: pkg/controller/podautoscaler/horizontal.go [hunk 1]
```diff
@@ -370,6 +273,134 @@ func (a *HorizontalController) reconcileKey(key string) error {
 	return a.reconcileAutoscaler(hpa)
 }
 
+// Computes the desired number of replicas for the specified metric of type ObjectMetricSourceType.
+func (a *HorizontalController) computeReplicasForObjectMetric(currentReplicas int32, metricSpec autoscalingv2.MetricSpec, hpa *autoscalingv2.HorizontalPodAutoscaler, selector labels.Selector, status *autoscalingv2.MetricStatus) (int32, time.Time, string, error) {
```
Comment (DirectXMan12) [inline pkg/controller/podautoscaler/horizontal.go]: name should imply that it's also setting status

**Review by DirectXMan12 — CHANGES_REQUESTED** [review]
minor nit inline, otherwise looks fine at a glance

**Comment by jbartosik** [comment]
I added a commit renaming helpers.
/retest

File: pkg/controller/podautoscaler/horizontal.go (hunk 1 above)
Comment (jbartosik) [reply pkg/controller/podautoscaler/horizontal.go]: Done.

File: pkg/controller/podautoscaler/horizontal.go [hunk 2]
```diff
@@ -274,7 +274,7 @@ func (a *HorizontalController) reconcileKey(key string) error {
 }
 
 // Computes the desired number of replicas for the specified metric of type ObjectMetricSourceType.
-func (a *HorizontalController) computeReplicasForObjectMetric(currentReplicas int32, metricSpec autoscalingv2.MetricSpec, hpa *autoscalingv2.HorizontalPodAutoscaler, selector labels.Selector, status *autoscalingv2.MetricStatus) (int32, time.Time, string, error) {
+func (a *HorizontalController) computeReplicasForObjectMetricSetStatus(currentReplicas int32, metricSpec autoscalingv2.MetricSpec, hpa *autoscalingv2.HorizontalPodAutoscaler, selector labels.Selector, status *autoscalingv2.MetricStatus) (int32, time.Time, string, error) {
```
Comment (DirectXMan12) [inline pkg/controller/podautoscaler/horizontal.go]: I'd maybe just call this `computeStatusForObjectMetric`  -- I think that's sufficiently descriptive (this name is a bit difficult to parse).

Comment (jbartosik) [reply pkg/controller/podautoscaler/horizontal.go]: Done.

File: pkg/controller/podautoscaler/horizontal.go [hunk 3]
```diff
@@ -370,6 +273,134 @@ func (a *HorizontalController) reconcileKey(key string) error {
 	return a.reconcileAutoscaler(hpa)
 }
 
+// Computes the desired number of replicas for the specified metric of type ObjectMetricSourceType.
```
Comment (DirectXMan12) [inline pkg/controller/podautoscaler/horizontal.go]: whoops, function Godoc should always start with the function name.

**Review by DirectXMan12 — CHANGES_REQUESTED** [review]
one last nit (godoc), then make sure this is squashed down to a series of logical commit, and then it's good to merge.

**Comment by DirectXMan12** [comment]
/lgtm
/approve

for future reference, I'd prefer to have an actual commit message body, even if it's just 

> computeReplicas for metrics is currently a bit long, and thus unwieldy to work with.
> This splits it up by type of metric.

**Pull request merged**